### PR TITLE
LPS-203279 Avoid null pointer exception

### PR DESCRIPTION
**Pull request by mateussgn**
Related issue: https://liferay.atlassian.net/browse/LPS-203279

### CODE REVIEW

File: modules/apps/object/object-service/src/main/java/com/liferay/object/internal/validation/rule/UniqueCompositeKeyObjectValidationRuleEngineImpl.java [hunk 1]
```diff
@@ -139,7 +139,9 @@ private Predicate _getPredicate(
 				Map<String, Object> objectFieldProperties =
 					(Map<String, Object>)entryValues.get(objectField.getName());
 
-				value = String.valueOf(objectFieldProperties.get("key"));
+				if (objectFieldProperties != null) {
```
Comment (guilhermedcamacho) [inline modules/apps/object/object-service/src/main/java/com/liferay/object/internal/validation/rule/UniqueCompositeKeyObjectValidationRuleEngineImpl.java:142]: Hey @mateussgn , what happens if objectFieldProperties is null, is there a problem to pass value variable with null value to `ObjectEntrySearchUtil.getUniqueCompositeKeyObjectFieldPredicate((Column<?, Object>)column, objectField.getDBType(), value)`?

Comment (guilhermedcamacho) [inline modules/apps/object/object-service/src/main/java/com/liferay/object/internal/validation/rule/UniqueCompositeKeyObjectValidationRuleEngineImpl.java:142]: Hey @mateussgn , what happens if objectFieldProperties is null, is there a problem to pass value variable with null value to `ObjectEntrySearchUtil.getUniqueCompositeKeyObjectFieldPredicate((Column<?, Object>)column, objectField.getDBType(), value)`?

**Comment by mateussgn** [comment]
ci:test:relevant

**Comment by mateussgn** [comment]
ci:test:sf

**Comment by mateussgn** [comment]
ci:test:object

File: modules/apps/object/object-service/src/main/java/com/liferay/object/internal/validation/rule/UniqueCompositeKeyObjectValidationRuleEngineImpl.java (hunk 1 above)
Comment (mateussgn) [reply modules/apps/object/object-service/src/main/java/com/liferay/object/internal/validation/rule/UniqueCompositeKeyObjectValidationRuleEngineImpl.java:142]: objectFieldProperties will just exist for picklist, that is why is inside an if statement.
In this case the value of a picklist it is inside of an object nested to objectFieldProperties.
If objectFieldProperties is null, then the value variable still filled with null.
value could be null if you see in line 133, it has been initialized with null.

Comment (guilhermedcamacho) [reply modules/apps/object/object-service/src/main/java/com/liferay/object/internal/validation/rule/UniqueCompositeKeyObjectValidationRuleEngineImpl.java:142]: My point is, is there a case where the field is of type picklist and it doesn't have objectFieldProperties? For me it seems to be the case for this NPE bug.
I understand that objectFieldProperties will only exist for picklist, but this was already being checked in an if statement for ObjectFieldConstants.BUSINESS_TYPE_PICKLIST.

**Review by guilhermedcamacho — APPROVED** [review]

**Pull request closed without merging**